math/matrix/equalmatrix.go

```diff
@@ -0,0 +1,15 @@
+package matrix
+
+func MatrixEqual[T comparable](mat1, mat2 [][]T) bool {
```

Comment:
You seem to be using `MatrixEqual` and `reflect.DeepEqual`. Be consisted and use a single function for equality checks.